### PR TITLE
Added opencv hand gesture recognition

### PR DESCRIPTION
**Pull request by isaacvandor**
Uses background subtraction, contours, and convexity to do real-time hand tracking

### CODE REVIEW

File: recognition.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+        blur = cv2.GaussianBlur(gray, (blurValue, blurValue), 0)
+        #cv2.imshow('blurred', blur) # show blur image
+        ret, thresh = cv2.threshold(blur, threshold, 255, cv2.THRESH_BINARY)
+        #cv2.imshow('binary', thresh) # show binary image
+
+
+        # get the coutours
+        new_threshold = copy.deepcopy(thresh)
+        img2,contours, hierarchy = cv2.findContours(new_threshold, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
```
Comment (Utsav22G) [inline recognition.py]: What does cv2.CHAIN_APPROX_SIMPLE do?

File: recognition.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
+import cv2
+import numpy as np
+import copy
+import math
+
+# parameters
+capture_region_x=0.5  # roi x start point
+capture_region_y=0.8  # roi y start point
+threshold = 60  #  threshold
+blur_value = 41  # GaussianBlur parameter
```
Comment (Utsav22G) [inline recognition.py]: did you try using different values of sigma for Gaussian Blur? in my experience, different values of sigma can have different effects on the edge blur. Also try bilateral blur.

File: recognition.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        img = removeBG(videostream)
+        img = img[0:int(capture_region_y * videostream.shape[0]),
+                    int(capture_region_x * videostream.shape[1]):videostream.shape[1]]  # clip the ROI
+        #cv2.imshow('mask', img) #show mask image
+
+        # convert the image into binary image
+        gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+        blur = cv2.GaussianBlur(gray, (blurValue, blurValue), 0)
+        #cv2.imshow('blurred', blur) # show blur image
+        ret, thresh = cv2.threshold(blur, threshold, 255, cv2.THRESH_BINARY)
```
Comment (Utsav22G) [inline recognition.py]: Try Otsu's threshold in addition to Binary?

File: recognition.py (hunk 1 above)
Comment (isaacvandor) [reply recognition.py]: "Do you need all the points on the line to represent that line? No, we need just two end points of that line. This is what cv2.CHAIN_APPROX_SIMPLE does." From https://docs.opencv.org/3.1.0/d4/d73/tutorial_py_contours_begin.html

File: recognition.py (hunk 3 above)
Comment (isaacvandor) [reply recognition.py]: Yeah we could, this isn't meant to be final...just to work

File: recognition.py (hunk 2 above)
Comment (isaacvandor) [reply recognition.py]: Yeah we can mess around with this a bit....there's also user thresholding built-in

**Pull request merged**